### PR TITLE
Build refactoring

### PR DESCRIPTION
**Pull request by reutshal**
## Status
Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/20330

## Description
- the build for instance tests has its own trigger. only runs on container number 0.
- unit-tests are running only in container number 1.
- fixed some issues in `instance_notifier` script

## Does it break backward compatibility?
   - No

## Must have
- [ ] Tests
- [ ] Documentation (with link to it)
- [X] Code Review

### CODE REVIEW

**Comment by reutshal** [comment]
This is a link to the INSTANCE_TESTS build: https://circleci.com/gh/demisto/content/35456#tests/containers/0

File: Tests/instance_notifier.py [hunk 1]
```diff
@@ -124,7 +113,7 @@ def slack_notifier(slack_token, secret_conf_path, server, user, password, build_
 
 if __name__ == "__main__":
     options = options_handler()
-    if options.nightly:
+    if options.instance_tests:
         with open('./Tests/instance_ips.txt', 'r') as instance_file:
```
Comment (yaakovi) [inline Tests/instance_notifier.py]: switch to using`env_results.json`

Comment (yaakovi) [reply Tests/instance_notifier.py]: also remove the condition on line 99 and then trigger the "instance tests" build again

File: .circleci/config.yml [hunk 2]
```diff
@@ -74,61 +92,100 @@ jobs:
...
+                echo "Skipping - running only in container number 0"
+                exit 0
+            fi
             if [ -n "${GITHUB_TOKEN}" ] ;
               then
                 python3 release_notes.py $CONTENT_VERSION $GIT_SHA1 $CIRCLE_BUILD_NUM $SERVER_VERSION --github-token $GITHUB_TOKEN
 
               else
-                python3 release_notes.py $CONTENT_VERSION $GIT_SHA1 $CIRCLE_BUILD_NUM $SERVER_VERSION
+                 python3 release_notes.py $CONTENT_VERSION $GIT_SHA1 $CIRCLE_BUILD_NUM $SERVER_VERSION
```
Comment (yaakovi) [inline .circleci/config.yml]:
```suggestion
                python3 release_notes.py $CONTENT_VERSION $GIT_SHA1 $CIRCLE_BUILD_NUM $SERVER_VERSION
```

File: .circleci/config.yml [hunk 3]
```diff
@@ -74,61 +92,100 @@ jobs:
...
-              python ./Tests/scripts/validate_files.py -c true -b $CHECK_BACKWARD
+            python ./Tests/scripts/validate_files.py -c true -b $CHECK_BACKWARD
       - run:
           name: Secrets
           when: always
           command: |
-              python3 ./Tests/scripts/hook_validations/secrets.py -c true
+            if [ -n "${INSTANCE_TESTS}" ] || [ $CIRCLE_NODE_INDEX -ne 0 ] ;
+              then
+                echo "Skipped Secrets step"
```
Comment (yaakovi) [inline .circleci/config.yml]: write the reason of skiping instead of just skipping

**Comment by reutshal** [comment]
No changelog or yml changes, adding the docs-approved label

**Pull request merged**